### DISCUSSION TITLE
how to inject window object(for example, window.location) into the rhino execution

### DISCUSSION THREAD

**Discussion by orzyoung** [mozilla/rhino#1487] [Q&A] [answered]
I'm new to rhino, we have the logic to use window.location object, but have no idea how the value can be injected during the execution

**Comment by p-bakker** [comment, marked as answer]
Assuming you know that window.location is a (clientside) browser thing, whereas Rhino is a serverside JavaScript that doesn't know anything about browser-specific APIs: if you're embedding rhino, you can expose custom host objects, so you can inject an object with a location property under the 'window' name. See our documentation on embedding for more details

**Comment by rbri** [comment]
@orzyoung maybe it is worth to have a look at HtmlUnit (https://github.com/HtmlUnit/htmlunit) that implements a whole browser including javascript support based on (a slightly modified and enhanced) Rhino.
